src/Security/SimpleIdentity.php

```diff
@@ -53,7 +53,12 @@ public function __construct($id, $roles = null, iterable $data = null)
 	 */
 	public function setId($id)
 	{
-		$this->id = is_numeric($id) && !is_float($tmp = $id * 1) ? $tmp : $id;
+		if (is_numeric($id) && !is_float($tmp = $id * 1)) {
+			$id = $tmp;
+		} elseif (!is_scalar($id) || is_bool($id)) {
```

Comment:
I think better is to checked whether it is `(!is_string($id) && !is_int($id))`

Reply:
@dg Fixed!